src/Corax/Utils/EntryTermsReader.cs

```diff
@@ -219,9 +230,10 @@ private void HandleRegularTerm(long termContainerId)
             TermId = (termContainerId >> 8) & ~0b111;
         }
 
-        if ((termContainerId & EntryTermsWriter.NullMarker) != 0 ) // null value
+        if ((TermId % Voron.Global.Constants.Storage.PageSize) == 0 && _rootPages.ToSpan().BinarySearch(TermId / Voron.Global.Constants.Storage.PageSize) >= 0) // null value
```

Comment:
Wouldn't this be better as a `HashSet`?

Reply:
Why? We also use the reader for `UnaryMatches`/`facets`/`morelikethis`, so this list can be created a lot.` NativeList` gives us little GC pressure in comparison to `HashSet,` right?